Frameworks/SPMySQLFramework/Source/SPMySQLArrayAdditions.h

```diff
@@ -0,0 +1,27 @@
+//
+//  SPMySQLArrayAdditions.h
+//  Sequel Ace
+//
+//  Created by James on 21/1/2021.
+//  Copyright © 2021 Sequel-Ace. All rights reserved.
+//
+
+
+@interface NSArray (SPMySQLArrayAdditions)
+
+/**
+ * Variant of objectAtIndex: that avoids the "index out of bounds" exception by
+ * just returning nil instead.
+ *
+ * @warning This method is NOT thread-safe.
+ * @param idx  An index
+ * @return The object located at index or nil.
+ */
+- (nullable id)SPsafeObjectAtIndex:(NSUInteger)idx;
```

Comment:
I believe the other "safe" methods aren't prefixed with SP. Why prefix these with SP?

Reply:
They are in the framework. If the same named methods are in the framework and main executable, then it builds, no errors, but at run-time you don't know which one will 'win', so I assed the prefix.

There is probably a better way to do this. Maybe just define the method signatures in the framework (so it builds), then at runtime it will find the methods from the executable?